### DISCUSSION TITLE
Can a non-modal Message Box Popup specify a parent?

### DISCUSSION THREAD

**Discussion by Ruikoto** [AvaloniaUI/Avalonia#17296] [Q&A] [answered]
After I upgraded from Avalonia 10 to 11, I found that in the new `IMsBox<T>` interface, the non-modal method `Task<T> ShowWindowAsync()` does not support specifying a parent control, whereas the other two methods that do support specifying a parent control are both modal dialogs. I want my multi-window application to display a non-modal dialog in front of a specific window. How can I achieve this?

```
namespace MsBox.Avalonia.Base
{
  public interface IMsBox<T>
  {
    Task<T> ShowAsync();

    Task<T> ShowWindowAsync();

    Task<T> ShowWindowDialogAsync(Window owner);

    Task<T> ShowAsPopupAsync(ContentControl owner);

    Task<T> ShowAsPopupAsync(Window owner);

    string InputValue { get; }
  }
}
```

**Comment by thevortexcloud** [comment, marked as answer]
As far as I am aware `MsBox` is not an official Avalonia namespace. Some quick searches shows it's probably this library.

https://github.com/AvaloniaCommunity/MessageBox.Avalonia

I suggest you direct your questions there.

**Reply by Ruikoto** [reply]
I understand, thank you for your response.
